allocator.go

```diff
@@ -0,0 +1,57 @@
+package parquet
+
+import "github.com/segmentio/parquet-go/internal/unsafecast"
+
+type allocator struct{ buffer []byte }
+
+func (a *allocator) makeBytes(n int) []byte {
+	if free := cap(a.buffer) - len(a.buffer); free < n {
+		newCap := 2 * cap(a.buffer)
+		if newCap == 0 {
+			newCap = 4096
+		}
+		for newCap < n {
+			newCap *= 2
+		}
+		a.buffer = make([]byte, 0, newCap)
+	}
+
+	i := len(a.buffer)
+	j := len(a.buffer) + n
+	a.buffer = a.buffer[:j]
+	return a.buffer[i:j:j]
+}
+
+func (a *allocator) copyBytes(v []byte) []byte {
+	b := a.makeBytes(len(v))
+	copy(b, v)
+	return b
+}
+
+func (a *allocator) copyString(v string) string {
+	b := a.makeBytes(len(v))
+	copy(b, v)
+	return unsafecast.BytesToString(b)
+}
+
+func (a *allocator) reset() {
+	a.buffer = a.buffer[:0]
+}
+
+// rowAllocator is a memory allocator used to make copy of rows referencing
+// memory buffers that parquet-go does not have ownership of.
```

Comment:
I'd appreciate another sentence about why we can't just use the Go allocator for this

Reply:
Let me know how that reads https://github.com/segmentio/parquet-go/pull/445/commits/d2f5a1f9ca5dbfc8a553c8a535a09d00ffc54c87